render/mapping.go

```diff
@@ -463,6 +559,64 @@ func MapContainerImage2Name(n RenderableNode, _ report.Networks) RenderableNodes
 	return RenderableNodes{name: node}
 }
 
+// MapService2Name maps container images RenderableNodes to
+// RenderableNodes for each service name.
+//
+// This mapper is unlike the other foo2bar mappers as the intention
+// is not to join the information with another topology.  Therefore
+// it outputs a properly-formed node with labels etc.
+func MapService2Name(n RenderableNode, _ report.Networks) RenderableNodes {
+	if n.Pseudo {
+		return RenderableNodes{n.ID: n}
+	}
+
+	name, ok := n.Node.Metadata[kubernetes.ServiceName]
+	if !ok {
+		return RenderableNodes{}
+	}
+
+	node := NewDerivedNode(name, n)
+	node.LabelMajor = name
+	node.Rank = name
+	node.Node = n.Node.Copy() // Propagate NMD for pod counting.
+	return RenderableNodes{name: node}
+}
+
+// MapContainer2Pod maps container RenderableNodes to pod
+// RenderableNodes.
+//
+// If this function is given a node without a kubernetes_pod_id
+// (including other pseudo nodes), it will produce an "Unmanaged"
+// pseudo node.
+//
+// Otherwise, this function will produce a node with the correct ID
+// format for a container, but without any Major or Minor labels.
+// It does not have enough info to do that, and the resulting graph
+// must be merged with a container graph to get that info.
+func MapContainer2Pod(n RenderableNode, _ report.Networks) RenderableNodes {
+	// Propogate all pseudo nodes
+	if n.Pseudo {
+		return RenderableNodes{n.ID: n}
+	}
+
+	// Otherwise, if some some reason the container doesn't have a pod_id
+	// (maybe slightly out of sync reports), just drop it
+	id, ok := n.Node.Metadata["docker_label_io.kubernetes.pod.name"]
+	if !ok {
+		return RenderableNodes{}
+	}
+
+	// Add container-<id> key to NMD, which will later be counted to produce the minor label
+	result := NewRenderableNodeWith(id, "", "", id, n)
+	result.Node.Counters[containersKey] = 1
+	if s := strings.SplitN(id, "/", 2); len(s) == 2 {
+		result.LabelMajor = s[1]
+		result.Node.Metadata[kubernetes.Namespace] = s[0]
+		result.Node.Metadata[kubernetes.PodName] = s[1]
```

Comment:
There is a bug in k8s where addon pods on the master are not returned via the API. The containers are still labeled (so still show up in here). This is a workaround until the bug is fixed. Will add a comment.